frontend/lib/features/audio-player/audio_player_screen.widget.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:just_audio/just_audio.dart';
+import 'package:mbl/features/audio-player/custom_progress_bar.widget.dart';
+import 'package:mbl/features/audio-player/player_button.widget.dart';
+import 'package:mbl/features/widget/media_app_bar.widget.dart';
+import 'package:mbl/repository/models/position_data.model.dart';
+import 'package:mbl/themes/themes.dart';
+import 'package:rxdart/rxdart.dart';
+
+class AudioPlayerScreen extends StatefulWidget {
+  const AudioPlayerScreen({
+    super.key,
+    required this.title,
+    required this.speaker,
+    required this.audioUrl,
+    required this.coverUrl,
+    });
+
+  final String title;
+  final String speaker;
+  final String audioUrl;
+  final String coverUrl;
+
+  @override
+  State<AudioPlayerScreen> createState() => _AudioPlayerScreenState();
+}
+
+class _AudioPlayerScreenState extends State<AudioPlayerScreen> {
+ late AudioPlayer _player;
+  
+Stream<PositionData> get _positionDataStream =>
+  Rx.combineLatest3<Duration, Duration, Duration?, PositionData>(
+    _player.positionStream,
+    _player.bufferedPositionStream,
+    _player.durationStream,
+    (position, bufferedPosition, duration) => PositionData(
+      position,
+      bufferedPosition,
+      duration ?? Duration.zero,
+    ),
+  );
+
+ @override
+  void initState() {
+    super.initState();
+    init();    
+  }
+
+  void init() async {
+    _player = AudioPlayer();
+    await _player.setUrl("http://${dotenv.get('HOST')}${widget.audioUrl}");
+  }
+
+  @override
+  void dispose() {
+    _player.dispose();
+    super.dispose();
+  }
+
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    if (state == AppLifecycleState.paused) {
+      _player.stop();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: StandardColor.primary,
+      appBar: const MediaAppBar(),
+      body: Column(
+        crossAxisAlignment: CrossAxisAlignment.center,
+        mainAxisSize: MainAxisSize.max,
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: [
+          Image.network("http://${dotenv.get('HOST')}${widget.coverUrl}"),
```

Comment:
In general you want to avoid function calls within function calls. I'm not too sure how that rule correlates to flutter development. Instinctively I would create a string variable like coverFullUrl in the build method and then just put in the variable in the Image.network function call.